src/Detector/ImplicitTypeConversionDetector.php

```
@@ -18,23 +19,17 @@ final class ImplicitTypeConversionDetector implements DetectorInterface
-        // reference_code のような文字列型のカラムに数値を直接比較している場合を検出
-        if ($this->containsStringNumericComparison($condition)) {
-            return true;
+        $walker = new ExplainWalker();
+        foreach ($walker->tables($explainResult) as $table) {
+            // attached_condition の確認
+            if (! isset($table['attached_condition'])) {
+                continue;
+            }
+
+            // reference_code のような文字列型のカラムに数値を直接比較している場合を検出
+            if ($this->containsStringNumericComparison((string) $table['attached_condition'])) {
```

PHPStan (2.2.1)

[error] Cannot cast mixed to string.


(cast.string)